standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -8349,8 +8353,12 @@ public long get_latest_txnid_in_conflict(long txnId) throws MetaException {
 
   @Override
   public void commit_txn(CommitTxnRequest rqst) throws TException {
+    boolean isReplayedReplTxn = rqst.isSetReplPolicy() && rqst.isSetTxn_type() &&
+            rqst.getTxn_type() == TxnType.REPL_CREATED;
+    boolean isHiveReplTxn = rqst.isSetReplPolicy()
+            && (!rqst.isSetTxn_type() || (rqst.isSetTxn_type() && rqst.getTxn_type() != TxnType.REPL_CREATED));
```

Comment:
Could you please explain what are you trying to check?

Reply:
It can be simplified as: 
boolean isHiveReplTxn = rqst.isSetReplPolicy() &&  rqst.getTxn_type() != TxnType.REPL_CREATED;

Just that for HiveReplTxn, replPolicy would be contain the dbName  and txnType would not be REPL_CREATED.

Reply:
In stead of  rqst.getTxn_type() != TxnType.REPL_CREATED you can have an affirmation on default type